tools/bench_deepgemm_stream.py

```
@@ -0,0 +1,426 @@
+            pass
+        new_times.append((time.perf_counter() - t0) * 1e6)  # μs
+
+    med_old = statistics.median(old_times)
+    med_new = statistics.median(new_times)
+    p99_old = sorted(old_times)[int(0.99 * len(old_times))]
+    p99_new = sorted(new_times)[int(0.99 * len(new_times))]
+
+    print(f"  stream_sync + device_sync : median={med_old:7.2f} μs  p99={p99_old:7.2f} μs")
+    print(f"  ExternalStream ctx enter  : median={med_new:7.2f} μs  p99={p99_new:7.2f} μs")
+    print(f"  (ExternalStream is ~{med_new - med_old:.1f} μs more in this micro-bench because")
+    print(f"   syncs return instantly with idle GPU. See GEMM bench for real-world impact.)")
```

Flake8 (7.3.0)

[error] f-string is missing placeholders

(F541)